Azure/sonic-mgmt
# Vnet vxlan extra RIF enhancement

### Description of PR


Summary:
Allow the VNET VxLAN test to be run with more than one RIF (VLAN interface) per VNET.

### Type of change




### Approach
#### What is the motivation for this PR?
The existing VNET VxLAN test only allows for 1 VLAN interface per VNET. Adding this enhancement allows for testing of more scenarios.
#### How did you do it?
* Add `num_intf_per_vnet` CLI parameter (defaults to 1)
* Modify `vnet_config.j2` to create additional VLAN interfaces starting at VLAN ID 2000
	* The first VLAN interface in each VNET is still created in the range starting at 3001 and should not be affected by this change.
    * All interfaces in the same VNET share the same port.
    * E.g. for 4 VLAN interfaces per VNET, VNET 1 will contain VLAN interfaces 3001, 2000, 2001, and 2002 on Ethernet28. VNET 2 will contain 3002, 2003, 2004, and 2005 on Ethernet32
* In Serv2Serv scenario in the PTF test, look for other VLANs in the same VNET and send packet to each to verify connectivity.
* Send ARP replies to all VLANs on a given port in `arp_responder.py`

#### How did you verify/test it?
* Run the test with default parameters (1 VLAN interface per VNET) and verify all scenarios pass
* Run the test with more than 1 interface per VNET and verify all scenarios pass
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 

`./run_tests.sh -n <testbed> -i <inventory> -u -c vxlan/test_vnet_vxlan.py -e --disable_loganalyzer -e --num_intf_per_vnet=<x>`
